src/main/webapp/app/lecture/lecture.component.ts

```diff
@@ -111,6 +120,17 @@ export class LectureComponent implements OnInit {
         this.applyFilters();
     }
 
+    sortRows() {
+        //Sort alphabetically for title and description
+        if (this.predicate == 'title')
+            this.filteredLectures = this.filteredLectures.sort((a: Lecture, b: Lecture) => (this.ascending ? a.title!.localeCompare(b.title!) : b.title!.localeCompare(a.title!)));
+        else if (this.predicate == 'description')
+            this.filteredLectures.sort((a: Lecture, b: Lecture) =>
+                this.ascending ? (a.description || '').localeCompare(b.description || '') : (b.description || '').localeCompare(a.description || ''),
+            );
+        else this.sortService.sortByProperty(this.filteredLectures, this.predicate, this.ascending);
```

Comment:
Why can't you always use the sortService? It can also take in an arbitrary string that specifies what attribute should be used for sorting and it also handles null/undefined cases

Reply:
As far as I understand sortService sorts strings alphanumerically (numbers followed by upper case letters and then lower case letters). In case that title or description starts with a lowercase letter the sorting result for these wont be alphabetical and can be misleading.

Reply:
@JohannesStoehr Would it make sense to adapt this in general? I just confirmed the behaviour for sorting exercises.
I would expect
0, 1, 2, ..., 9, A, a, B, b, ..., Y, y, Z, z
instead of
0, 1, 2, ..., 9, A, B, C, ..., Y, Z, a, b, c, ..., y, z

Reply:
I can't think of any occurrence where we would want to first have all capital letters. So a change in the service should be possible

Reply:
@basak-akan Would you mind adapting this, then? I think here or in another PR is equally fine.

Reply:
Sure, I will adapt this here👍